presto-orc/src/main/java/com/facebook/presto/orc/OrcWriter.java

```diff
@@ -885,20 +894,51 @@ private static <T> List<T> toDenseList(Map<Integer, T> data, int expectedSize)
         return denseList.build();
     }
 
-    private static List<ColumnStatistics> toFileStats(List<List<ColumnStatistics>> stripes)
+    private void collectStreamSizeForColumnStatistics(StreamDataOutput stream)
+    {
+        requireNonNull(stream, "stream is null");
+        int node = stream.getStream().getColumn();
+        long size = nodeSizes.getOrDefault(node, 0L);
+        size += stream.size();
+        nodeSizes.put(node, size);
+    }
+
+    private static Int2LongMap rollupNodeSizes(Int2LongMap nodeSizes, List<OrcType> orcTypes)
```

Comment:
Storage size of a parent node in a complex type is self size + children sizes, that's why we need to recursively rollup stats for complex types.